### PR TITLE
Nipun/add ssv migrate flag

### PR DESCRIPTION
**Pull request by fighting-dreamer**

### CODE REVIEW

File: shared/services/config/stader-config.go [hunk 1]
```diff
@@ -159,20 +161,30 @@ func LoadFromFile(path string) (*StaderConfig, error) {
...
-
+	// to be removed : start-02
+	fmt.Println("isSSVMode : ", cfg.IsSSVMode)
+	// to be removed : end
 	return cfg, nil
 
 }
 
 // Creates a new Stader configuration instance
-func NewStaderConfig(staderDir string, isNativeMode bool) *StaderConfig {
```
Comment (batphonghan) [inline shared/services/config/stader-config.go:175]: Do not changes this API

File: shared/services/config/stader-config.go [hunk 2]
```diff
@@ -88,7 +90,7 @@ type StaderConfig struct {
 
 	// Metrics settings
 	EnableMetrics           config.Parameter `yaml:"enableMetrics,omitempty"`
-	ExposeGuardianPort      config.Parameter `yaml:"enableMetrics,omitempty"`
+	ExposeGuardianPort      config.Parameter `yaml:"exposeGuardianPort,omitempty"`
```
Comment (batphonghan) [inline shared/services/config/stader-config.go:93]: Why you change this, this's not backward compatible change

File: shared/services/config/stader-config.go [hunk 3]
```diff
@@ -159,20 +161,30 @@ func LoadFromFile(path string) (*StaderConfig, error) {
 	if err := yaml.Unmarshal(configBytes, &settings); err != nil {
 		return nil, fmt.Errorf("could not parse settings file: %w", err)
 	}
+	// to be removed : start-01
```
Comment (batphonghan) [inline shared/services/config/stader-config.go:164]: if you think this is unnecessary, remove it

File: shared/services/config/stader-config.go [hunk 4]
```diff
@@ -72,6 +73,7 @@ type StaderConfig struct {
 	StaderDirectory string `yaml:"-"`
 
 	IsNativeMode bool `yaml:"-"`
+	IsSSVMode    bool `yaml:"-"`
```
Comment (batphonghan) [inline shared/services/config/stader-config.go:76]: Why using `-`, you should put it in `StaderNodeConfig`

**Review by batphonghan — CHANGES_REQUESTED** [review]

File: shared/services/config/stader-config.go (hunk 1 above)
Comment (fighting-dreamer) [reply shared/services/config/stader-config.go:175]: any reason ?

File: shared/services/config/stader-config.go (hunk 2 above)
Comment (fighting-dreamer) [reply shared/services/config/stader-config.go:93]: yea,  you mean for users who have not set exposeGuardianPort config. I see service still start without this config set.

this is something we should fix
the `stader-cli service get-config-yaml` breaks btw

File: shared/services/config/stader-config.go (hunk 1 above)
Comment (batphonghan) [reply shared/services/config/stader-config.go:175]: To keep it simple, do not treat new flag as special.

File: shared/services/config/stader-config.go (hunk 2 above)
Comment (batphonghan) [reply shared/services/config/stader-config.go:93]: if exiting node had set `ExposeGuardianPort`  to `true`, which's the value you load in new code?

File: shared/services/config/stader-config.go (hunk 4 above)
Comment (batphonghan) [reply shared/services/config/stader-config.go:76]: `stader-config` or `stader-node-config` is not big things you can choose the place. However, please refrain to use `-`, you will need to manually serialize it.

File: shared/services/config/stader-config.go (hunk 3 above)
Comment (fighting-dreamer) [reply shared/services/config/stader-config.go:164]: removing the whole printing og config object, keeping the printing of ssv-flag, will remove after successful dev testing.

File: shared/services/config/stader-config.go (hunk 4 above)
Comment (fighting-dreamer) [reply shared/services/config/stader-config.go:76]: setting a name will lead to handle default cases of yaml files that are not setting this for backward compatibility.
keeping it to de-serialize from the user settings where user is expected to manually add this to use ssv-mode for now.

File: shared/services/config/stader-config.go (hunk 1 above)
Comment (fighting-dreamer) [reply shared/services/config/stader-config.go:175]: ssv mode flag is set in stader config, if we dont add this is NewStaderConfig func, we break some calls of this method in other places.

**Pull request closed without merging**